Sources/MapboxMaps/Style/Types/ExpressionOptions.swift

```diff
@@ -47,12 +47,21 @@ public struct FormatOptions: Codable, Equatable, ExpressionArgumentConvertible {
 
     /// Applies a scaling factor on text-size as specified by the root layout property.
     public var fontScale: Double?
+    /// Applies a scaling factor on text-size as specified by the root layout property.
+    /// If specified, this value will override ``fontScale`` when encoding.
+    public var fontScaleExpression: Expression?
```

Comment:
Hi Mai, thank you for picking it up!

It's good workaround, but I have another idea - what it if we use computed properties for backward compatibility and `Value<T>` as backing store:
```swift
public var fontScaleValue: Value<Double>?
public var fontScale: Double? {
    get { fontScaleValue?.asConstant }
    set { fontScaleValue = newValue.map { .constant($0) } }
}

internal enum CodingKeys: String, CodingKey {
    case fontScaleValue = "font-scale"
}

extension Value {
    var asConstant: T? {
        switch self {
        case let .constant(c): return c
        case .expression: return nil
        }
    }
}
```

It can be better because:
- there is a single source of truth, the value in `fontScale` and `fontScaleValue` can't diverge after user modifications.
- no need to write custom encoders/decoders
- for v11 we can just remove old `fontScale` property and rename `fontScaleValue` to `fontScale`. If customers use `fontScaleValue` it's also trivial to migrate 

What do you think?